packages/shared/src/components/modals/OnboardingModal.tsx

```diff
@@ -4,49 +4,79 @@ import React, {
   ReactElement,
   ReactNode,
   useContext,
+  useEffect,
   useMemo,
   useRef,
   useState,
 } from 'react';
 import { ModalProps } from './StyledModal';
-import SteppedModal from './SteppedModal';
+import SteppedModal, { getForwardLabel } from './SteppedModal';
 import ThemeOnboarding from '../onboarding/ThemeOnboarding';
 import FilterOnboarding from '../onboarding/FilterOnboarding';
 import LayoutOnboarding from '../onboarding/LayoutOnboarding';
 import IntroductionOnboarding from '../onboarding/IntroductionOnboarding';
 import FeaturesContext from '../../contexts/FeaturesContext';
 import { OnboardingStep } from '../onboarding/common';
-import { LoginTrigger } from '../../lib/analytics';
+import {
+  AnalyticsEvent,
+  LoginTrigger,
+  ScreenValue,
+  TargetType,
+} from '../../lib/analytics';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { OnboardingMode } from '../../graphql/feed';
+import { OnboardingVersion } from '../../lib/featureValues';
 import DiscardActionModal from './DiscardActionModal';
 
 const INTRODUCTION_ADDITIONAL_STEP = 1;
 
 interface OnboardingModalProps extends ModalProps {
-  trigger?: string;
+  mode?: OnboardingMode;
   onRegistrationSuccess?: () => void;
 }
 
+const getScreen = (onboardingSteps: OnboardingStep[], stepAtNext: number) => {
+  if (stepAtNext === 0) {
+    return ScreenValue.Intro;
+  }
+
+  const index = stepAtNext - INTRODUCTION_ADDITIONAL_STEP;
+  const screen = onboardingSteps[index];
+  return screen;
+};
+
 const backCopy = ['Skip', 'Close', 'Back', 'Back'];
 const nextCopy = ['Continue'];
 
 function OnboardingModal({
   onRegistrationSuccess,
   onRequestClose,
+  mode = OnboardingMode.Manual,
   ...props
 }: OnboardingModalProps): ReactElement {
+  const { trackEvent } = useContext(AnalyticsContext);
   const [isClosing, setIsClosing] = useState(false);
   const topics = useRef({});
   const [invalidMessage, setInvalidMessage] = useState<string>(null);
-  const { onboardingSteps, onboardingMinimumTopics } =
+  const { onboardingVersion, onboardingSteps, onboardingMinimumTopics } =
     useContext(FeaturesContext);
   const [step, setStep] = useState(0);
 
-  const onCloseConfirm = (
-    e: MouseEvent | KeyboardEvent,
-    stepAtClose?: number,
-  ) => {
+  const onCloseConfirm = (e: MouseEvent | KeyboardEvent) => {
+    if (
+      mode === OnboardingMode.Auto &&
+      onboardingVersion === OnboardingVersion.V2
+    ) {
```

Comment:
Why do you apply this condition for tracking?

Reply:
The first one is the same as the previous implementation, the second one is since we already have a tracking event in our common custom hook but only designed for v1 due to their differences in the behavior of closing the modal.